src/xtgeo/grid3d/_egrid.py

```diff
@@ -254,7 +261,9 @@ def _check_xtgeo_compatible(self):
             )
 
     @property
-    def dimensions(self) -> tuple[int, int, int]:
+    def dimensions(self) -> tuple[int, int, int] | None:
+        if self.grid_head is None:
```

Comment:
Since `grid_head` is no longer `Optional`, this check is no longer necessary. Likely, the return type need not include None any more.